libs/boost/patches/001-uclibc-asio.patch

```diff
@@ -29,40 +29,40 @@ index 38d4b2a61..e16cc8b00 100644
  #include <sys/stat.h>
  #include <sys/types.h>
  #include <fcntl.h>
--#if __GLIBC__ == 2 && __GLIBC_MINOR__ < 8
-+#if __GLIBC__ == 2 && __GLIBC_MINOR__ < 2
+-#if __GLIBC__ == 2 && __GLIBC_MINOR__ < 8 && !defined(__UCLIBC__)
++#if __GLIBC__ == 2 && __GLIBC_MINOR__ < 2 && !defined(__UCLIBC__)
```

Comment:
Why this change?

Reply:
@neheb The patch needed to be updated otherwise it would fail to be applied. Only that extra macro verification was added to the source. Creating a new patch was more troublesome. As so, I just updated directly the patch.

Reply:
That makes no sense. __GLIBC_MINOR__ is still 2 under uClibc-ng. The change from 8 to 2 is what should fix it.

Out of curiosity, did you run make package/boost/refresh ?

Reply:
No. My usual procedure is to `wget` the master, script update & install the packages, update the makefile, go to menu config and enable all packages from boost (+ python3) and build all. Check builded packages and if all goes well without errors, I push the changes. If not, check errors, make necessary adjustments, then clean and compile again.

It has served me well in the past.
Btw, `&& !defined(__UCLIBC__)` was added in the boost sources. Is OpenWRT even supporting uclibc? I don't recall seeing it in the advance toolchain options for the Raspberry Pi 4 target.

Reply:
I see now. This patch can be removed. Upstream fixed the issue differently it seems.

make package/x/refresh will soon be added as a check. Although it seems that without this patch, boost is patch free.

Reply:
Yep. I just also concluded that... Sorry about the mess. I was in automatic mode and didn't event checked the versions properly.

Reply:
I have that patch as a PR. Seems the ASIO maintainers don't look at them.

Reply:
Just removed the patch. Since there are no more patches, removed directory as well.